libcontainer/container_linux.go

```diff
@@ -163,8 +163,39 @@ func (c *linuxContainer) OCIState() (*specs.State, error) {
 	return c.currentOCIState()
 }
 
+func (c *linuxContainer) cgroupPathNotExist() bool {
+	if cgroups.IsCgroup2UnifiedMode() {
+		path, err := c.cgroupManager.GetUnifiedPath()
+		if err == nil {
+			if _, err := os.Stat(path); os.IsNotExist(err) {
+				return true
+			}
+		}
+	} else {
+		paths := c.cgroupManager.GetPaths()
+		if path := paths["device"]; path != "" {
+			if _, err := os.Stat(path); os.IsNotExist(err) {
+				return true
+			}
+		}
+	}
+	return false
+}
+
 func (c *linuxContainer) Processes() ([]int, error) {
-	pids, err := c.cgroupManager.GetAllPids()
+	var pids []int
+	status, err := c.currentStatus()
+	if err != nil {
+		return pids, err
+	}
+	if status == Stopped {
+		// for systemd cgroup, the unit's cgroup path will be auto removed if container's all processes exited
+		if check := c.cgroupPathNotExist(); check {
+			return pids, err
+		}
+	}
+
+	pids, err = c.cgroupManager.GetAllPids()
```

Comment:
wouldn't this already be able to have a check for `os.IsNotExist(err)` ? (Also wondering if `c.cgroupManager.GetAllPids()` should ignore `os.IsNotExist(err)` by default (if the path doesn't exist, just return an empty list of pids ?

Reply:
> wouldn't this already be able to have a check for `os.IsNotExist(err)` ?

Yes, but in fs cgroup driver, there is no error when the container is in `stopped` state. And I think we should not ignore `IsNotExist` error when it is not in `stopped` state or it has joined a new pid namespace without path.

> (Also wondering if `c.cgroupManager.GetAllPids()` should ignore `os.IsNotExist(err)` by default (if the path doesn't exist, just return an empty list of pids ?

I think we should not ignore `IsNotExist` error when it is not in `stopped` state or it has joined a new pid namespace without path.